satellite/satellitedb/nodeselection.go

```diff
@@ -0,0 +1,323 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedb
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/zeebo/errs"
+
+	"storj.io/common/pb"
+	"storj.io/common/storj"
+	"storj.io/private/version"
+	"storj.io/storj/satellite/overlay"
+)
+
+func (cache *overlaycache) SelectStorageNodes(ctx context.Context, totalNeededNodes, newNodeCount int, criteria *overlay.NodeCriteria) (nodes []*overlay.SelectedNode, err error) {
+	defer mon.Task()(&ctx)(&err)
+	if totalNeededNodes == 0 {
+		return nil, nil
+	}
+
+	needNewNodes := newNodeCount
```

Comment:
should there be a check to make sure that `newNodeCount` is less than `totalNeededNodes`?

Reply:
that's a good idea!